check-update-security.sh

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env bash
+
+set -eio pipefail
+
+usage() {
+    echo "Usage:"
+    echo "  $0 AMI_PLATFORM"
+    echo "Example:"
+    echo "  $0 al2_arm"
+    echo "AMI_PLATFORM Must be one of: al, al2, al2_arm, al2023, al2023_arm"
+}
+
+error_msg() {
+    local msg="$1"
+    echo "ERROR: $msg"
+}
+
+# Paths to get the ami ids from ssm params
+AL1_PATH="/aws/service/ecs/optimized-ami/amazon-linux/recommended"
+AL2_PATH="/aws/service/ecs/optimized-ami/amazon-linux-2/recommended"
+AL2_ARM_PATH="/aws/service/ecs/optimized-ami/amazon-linux-2/arm64/recommended"
+AL2023_PATH="/aws/service/ecs/optimized-ami/amazon-linux-2023/recommended"
+AL2023_ARM_PATH="/aws/service/ecs/optimized-ami/amazon-linux-2023/arm64/recommended"
+
+# Indicates that an update exists
+UPDATE_EXISTS_CODE="100"
+# Indicates that a wait operation failed
+WAIT_FAIL_CODE="255"
+# Indicates success
+SUCCESS_CODE="0"
+
+# In case of failure, terminate instance
+failure_cleanup() {
+    terminate_out=$(aws ec2 terminate-instances --instance-ids $instance_id)
+}
+
+check_wait_response() {
+    local wait_response=$1
+    if [ "$wait_response" -eq "$WAIT_FAIL_CODE" ]; then
+        error_msg "Failed to launch instance, instance timeout"
+        exit 1
+    fi
+}
+
+platform=$1
+if [ -z "$platform" ]; then
+    error_msg "Must specify an AMI platform"
+    usage
+    exit 1
+fi
+if [ -z "$IAM_INSTANCE_PROFILE_ARN" ]; then
+    error_msg "IAM_INSTANCE_PROFILE_ARN environment variable must exist"
+    exit 1
+fi
+
+# Get ecs-optimized ami's PATH in ssm parameters
+platform=$1
+skip_user_data=0
+instance_type="c5.large"
+install_and_start_ssm_agent=0
+case "$platform" in
+"al1")
+    ami_path=$AL1_PATH
+    install_and_start_ssm_agent=1 # AL1 Requires Installation of SSM Agent
+    ;;
+"al2")
+    ami_path=$AL2_PATH
+    ;;
+"al2_arm")
+    ami_path=$AL2_ARM_PATH
+    instance_type="c6g.medium"
+    ;;
+"al2023")
+    ami_path=$AL2023_PATH
+    skip_user_data=1
+    ;;
+"al2023_arm")
+    ami_path=$AL2023_ARM_PATH
+    skip_user_data=1
+    instance_type="c6g.medium"
+    ;;
+*)
+    error_msg "Incorrect platform selection"
+    usage
+    exit 1
+    ;;
+esac
+
+# Query ssm to get latest ecs optimized ami
+ami_id=$(aws ssm get-parameters --names $ami_path --region us-west-2 | jq -r '.Parameters[0].Value' | jq -r '.image_id')
```

Comment:
Is this only to be run in us-west-2? This might be related to how we setup pipeline or DJS. Have we set the region to be us-west-2?

Reply:
yup, since this is only a check to see whether updates are necessary to the ami. The release will cause the updated ami to spread to every region anyway, so we only need to check one.

Reply:
> Is this only to be run in us-west-2?

Yes, the GH actions will trigger this script only in 1 region. 

Halit - Should we make this us-west-1? (we start building in us-west-1)